src/main/java/com/metamx/metrics/SigarUtil.java

```diff
@@ -58,9 +58,20 @@
     }
   }
 
+  /**
+   * SigarHolder class is initialized after SigarUtil, that guarantees that SIGAR = new Sigar() is executed after static
+   * block (which loads the library) of SigarUtil is executed. This is anyway guaranteed by JLS if SIGAR static field
+   * goes below the static block in textual order, but fragile e. g. if someone applies automatic reformatting and the
+   * static field is moved above the static block.
+   */
+  private static class SigarHolder
+  {
+    private static final Sigar SIGAR = new Sigar();
+  }
+
   public static Sigar getSigar()
   {
-    return new Sigar();
+    return SigarHolder.SIGAR;
```

Comment:
@logarithm seems that Sigar is not thread-safe indeed; so I reverted to status quo, caching only getPid(): https://github.com/metamx/server-metrics/pull/20/commits/53522e19a6f7dde9b5a41403ed852bbbc5a247fb